app/controllers/courses_controller.rb

```diff
@@ -560,14 +564,44 @@ def parse_roster_csv
                    section: row[10].to_s.chomp(" ") }
         cud = @currentCUDs.find do |cud|
           cud.user && cud.user.email == newCUD[:email]
+
+        if newCUD[:email].nil? || newCUD[:email].size == 0
```

Comment:
Should we be running all of these if/elsif checks manually? Originally, the error-checking was linked to the active record validations of User (see the line `validates :first_name, :last_name, :email, presence: true' in `app/models/user.rb'), and based on that, only first_name, last_name, email, are currently required, which I believe is a reasonable requirement. I feel like checking with if/elsif could also cause problems in the future, because if we ever need to change the required fields in the model, we now have to remember to change them here too.

Also, I haven't been able to run this because I'm currently having issues with Ruby, but it looks like it currently only shows the first error that shows up in the roster. Ideally, it should return all errors so that users don't have to resubmit to check if there are any other errors after they fix the first one.